mongodb/migrations/migrations_test.go

```diff
@@ -183,6 +183,176 @@ func TestMigrations_m004(t *testing.T) {
+					},
+				},
+			},
+		},
+		bson.M{
+			"_id": "id3",
+			"archives": bson.M{
+				"current": bson.M{
+					"job_id":     "job9",
+					"status":     1,
+					"created_at": 1000,
+				},
+				"history": bson.A{
+					bson.M{
+						"job_id":     "job8",
+						"job_status": 1,
+						"created_at": 1000,
+					},
+					bson.M{
+						"job_id":     "job7",
+						"status":     1,
+						"created_at": 1000,
+					},
+				},
+			},
+		},
+		bson.M{
+			"_id": "id4",
+			"archives": bson.M{
+				"current": bson.M{
+					"job_id":     "job12",
+					"job_status": 1,
+					"created_at": 1000,
+				},
+				"history": bson.A{
+					bson.M{
+						"job_id":     "job11",
+						"status":     1,
+						"created_at": 1000,
+					},
+					bson.M{
+						"job_id":     "job10",
+						"status":     1,
+						"created_at": 1000,
+					},
+				},
+			},
+		},
+		bson.M{
+			"_id": "id5",
+			"archives": bson.M{
+				"history": bson.A{
+					bson.M{
+						"job_id":     "job14",
+						"job_status": 1,
+						"created_at": 1000,
+					},
+					bson.M{
+						"job_id":     "job13",
+						"status":     1,
+						"created_at": 1000,
+					},
+				},
+			},
+		},
+		bson.M{
+			"_id": "id6",
+			"archives": bson.M{
+				"current": bson.M{
+					"job_id":     "job15",
+					"job_status": 1,
+					"created_at": 1000,
+				},
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	// Run up
+	err = migrate.NewMigrate(db, m005).Up(migrate.AllAvailable)
+	require.NoError(t, err)
+
+	count, err := db.Collection("bucketarchives").CountDocuments(ctx, bson.M{})
+	require.NoError(t, err)
+	assert.Equal(t, 6, int(count))
+	cursor, err := db.Collection("bucketarchives").Find(ctx, bson.M{})
+	require.NoError(t, err)
+	defer cursor.Close(ctx)
+
+	validateItem := func(item bson.M) {
+		require.NotEmpty(t, item["job_id"])
+		require.Contains(t, item["job_id"], "job")
+		require.Equal(t, 1000, int(item["created_at"].(int32)))
+		require.Nil(t, item["job_status"])
+		require.Equal(t, 1, int(item["status"].(int32)))
+	}
+
+	for cursor.Next(ctx) {
+		var item bson.M
+		require.NoError(t, cursor.Decode(&item))
+		require.NotEmpty(t, item["_id"])
+		require.NotEmpty(t, item["archives"])
+		if item["_id"] == "id1" || item["_id"] == "id2" || item["_id"] == "id3" || item["_id"] == "id4" {
+			require.NotNil(t, item["archives"].(bson.M)["current"])
+			require.NotNil(t, item["archives"].(bson.M)["history"])
+		}
+		if item["_id"] == "id5" {
+			require.Nil(t, item["archives"].(bson.M)["current"])
+			require.NotNil(t, item["archives"].(bson.M)["history"])
+		}
+		if item["_id"] == "id6" {
+			require.NotNil(t, item["archives"].(bson.M)["current"])
+			require.Nil(t, item["archives"].(bson.M)["history"])
+		}
+		if item["archives"].(bson.M)["current"] != nil {
+			validateItem(item["archives"].(bson.M)["current"].(bson.M))
+		}
+		if item["archives"].(bson.M)["history"] != nil {
+			for _, item := range item["archives"].(bson.M)["history"].(bson.A) {
+				validateItem(item.(bson.M))
+			}
+		}
```

Comment:
Validate all the contained archive infos under `current` and the `history` array.